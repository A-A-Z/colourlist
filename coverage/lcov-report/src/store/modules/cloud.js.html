<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/modules/cloud.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/store/modules</a> cloud.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">43.48% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.91% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/22</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { SAVE_STATES } from '@/constants'
import { db, auth, ProjectCollection } from '@/api/firebase.js'
import {
  SET_COLOURS,
  SET_PROJECT,
  SET_SAVE_STATE,
  SET_USER
} from '../mutation-types'
import { CONNECT, CREATE, LOAD, UPDATE } from '../action-types'
&nbsp;
const state = () =&gt; ({
  saveState: SAVE_STATES.MOUNTED,
  user: null
})
&nbsp;
const mutations = {
  [SET_SAVE_STATE] (state, newState) {
    state.saveState = newState
  },
&nbsp;
  [SET_USER] (state, userId) {
    state.user = userId
  }
}
&nbsp;
const actions = {
  [CONNECT]: ({ commit }) =&gt; {
    // sign into firebase anonymously
    auth.signInAnonymously()
      .catch(<span class="fstat-no" title="function not covered" >er</span>ror =&gt; {
<span class="cstat-no" title="statement not covered" >        commit(SET_SAVE_STATE, SAVE_STATES.ERROR)</span>
<span class="cstat-no" title="statement not covered" >        console.error('signInAnonymously error', error) </span>// TODO error hanlding
      })
&nbsp;
    // set up watcher for user login
    auth.onAuthStateChanged(user =&gt; {
      commit(SET_SAVE_STATE, SAVE_STATES.READY)
      commit(SET_USER, user)
    })
  },
&nbsp;
  [CREATE]: <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    /* TODO */
<span class="cstat-no" title="statement not covered" >    console.log('Create...')</span>
  },
&nbsp;
  async [LOAD] ({ commit }, projectId) {
<span class="cstat-no" title="statement not covered" >    commit(SET_SAVE_STATE, SAVE_STATES.LOADING)</span>
&nbsp;
    const { docs } = <span class="cstat-no" title="statement not covered" >await ProjectCollection.where('id', '==', projectId).get()</span>
<span class="cstat-no" title="statement not covered" >    if (docs.length) {</span>
      // project found
      const project = <span class="cstat-no" title="statement not covered" >docs[0].data()</span>
<span class="cstat-no" title="statement not covered" >      commit(`colourList/${SET_COLOURS}`, project, { root: true })</span>
<span class="cstat-no" title="statement not covered" >      commit(`project/${SET_PROJECT}`, { id: project.id, title: project.name }, { root: true })</span>
<span class="cstat-no" title="statement not covered" >      commit(SET_SAVE_STATE, SAVE_STATES.READY)</span>
    } else {
      // project not found
<span class="cstat-no" title="statement not covered" >      commit(SET_SAVE_STATE, SAVE_STATES.ERROR)</span>
<span class="cstat-no" title="statement not covered" >      console.error('Project not found') </span>// TODO error handling
    }
  },
&nbsp;
  [UPDATE]: <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    /* TODO */
<span class="cstat-no" title="statement not covered" >    console.log('Update...')</span>
  }
}
&nbsp;
export default {
  namespaced: true,
  state,
  mutations,
  actions
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 03 2019 15:27:50 GMT+1000 (Australian Eastern Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
