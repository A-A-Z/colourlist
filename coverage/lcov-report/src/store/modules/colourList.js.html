<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/modules/colourList.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/store/modules</a> colourList.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">48.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>20/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">26.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.15% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/39</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Patterns from '@/helpers/Patterns'
import GetColourName from '@/helpers/GetColourName'
import { DEFAULT_NEW_COLOUR_INPUT } from '@/constants'
import {
  ADD_COLOUR_TO_LIST,
  NEW_COLOUR_CHANGE,
  REMOVE_COLOUR_FROM_LIST,
  SET_ACTIVE_COLOUR,
  SET_COLOURS
} from '../mutation-types'
import {
  ON_NEW_COLOUR_INPUT
} from '../action-types'
&nbsp;
const state = () =&gt; ({
  activeColour: null,
  colours: [],
  newColourInput: DEFAULT_NEW_COLOUR_INPUT,
})
&nbsp;
const mutations = {
  [ADD_COLOUR_TO_LIST] (state) {
    let colours = state.colours.slice()
    colours.push(state.newColourInput)
    state.colours = colours
    state.newColourInput = DEFAULT_NEW_COLOUR_INPUT
  },
&nbsp;
  [NEW_COLOUR_CHANGE] (state, value) {
    state.newColourInput = value
  },
&nbsp;
  [REMOVE_COLOUR_FROM_LIST] (state, colour) {
    // filter out the passed colour
    state.colours = state.colours.filter(entry =&gt; entry !== colour)
&nbsp;
    // if colour is also the active colour then clear that
    <span class="missing-if-branch" title="if path not taken" >I</span>if (state.activeColour === colour) {
<span class="cstat-no" title="statement not covered" >      state.activeColour = null</span>
<span class="cstat-no" title="statement not covered" >      state.newColourInput = DEFAULT_NEW_COLOUR_INPUT</span>
    }
  },
&nbsp;
  [SET_ACTIVE_COLOUR] (state, colour) {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (state.activeColour === colour) {
<span class="cstat-no" title="statement not covered" >      state.activeColour = null</span>
<span class="cstat-no" title="statement not covered" >      state.newColourInput = DEFAULT_NEW_COLOUR_INPUT</span>
    } else {
      state.activeColour = colour
      state.newColourInput = colour
    }
  },
&nbsp;
  [SET_COLOURS] (state, { list }) {
    state.colours = list
  }
}
&nbsp;
const getters = {
  newColourHex: ({ newColourInput }, getters, { settings }) =&gt; {
    // return the active colour as formatted HEX
    const colour = (/^#/.test(newColourInput)) ? newColourInput : <span class="branch-1 cbranch-no" title="branch not covered" >`#${newColourInput}`</span>
    return settings.isLowercase ? colour.toLowerCase() : <span class="branch-1 cbranch-no" title="branch not covered" >colour.toUpperCase()</span>
  },
&nbsp;
  isNewColourValid: ({ newColourInput }) =&gt; Patterns.validColour.test(newColourInput),
&nbsp;
  colourNames: <span class="fstat-no" title="function not covered" >({</span> colours }) =&gt; {
    let names = <span class="cstat-no" title="statement not covered" >{}</span>
    let nameCount = <span class="cstat-no" title="statement not covered" >{}</span>
    let firsts = <span class="cstat-no" title="statement not covered" >{}</span>
&nbsp;
    // create names list from this.colours
    // (lets try to do this in one loop)
<span class="cstat-no" title="statement not covered" >    for (const colour of colours) {</span>
      let name = <span class="cstat-no" title="statement not covered" >GetColourName(colour)</span>
<span class="cstat-no" title="statement not covered" >      nameCount[name] = (nameCount[name] + 1) || 1</span>
&nbsp;
      // checking for duplicates
<span class="cstat-no" title="statement not covered" >      switch (nameCount[name]) {</span>
        case 1: // new colour name
<span class="cstat-no" title="statement not covered" >          names[colour] = name</span>
<span class="cstat-no" title="statement not covered" >          firsts[name] = colour</span>
<span class="cstat-no" title="statement not covered" >          break</span>
&nbsp;
        case 2: // name used once before
<span class="cstat-no" title="statement not covered" >          names[firsts[name]] = `${name} 1` </span>// rename the first
          /* falls through */
&nbsp;
        default: // name used more then once
<span class="cstat-no" title="statement not covered" >          names[colour] = `${name} ${nameCount[name]}`</span>
      }
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    return names</span>
  }
}
&nbsp;
const actions = {
  [ON_NEW_COLOUR_INPUT] ({ commit, getters }, value) {
    const oldValue = <span class="cstat-no" title="statement not covered" >getters.newColourHex</span>
&nbsp;
    // set new value
<span class="cstat-no" title="statement not covered" >    commit(NEW_COLOUR_CHANGE, value)</span>
&nbsp;
    // if new value is invalid then reset to old value
<span class="cstat-no" title="statement not covered" >    if (!Patterns.inputColour.test(value)) {</span>
<span class="cstat-no" title="statement not covered" >      commit(NEW_COLOUR_CHANGE, oldValue)</span>
    }
  }
}
&nbsp;
export default {
  namespaced: true,
  state,
  mutations,
  getters,
  actions
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 03 2019 15:27:50 GMT+1000 (Australian Eastern Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
