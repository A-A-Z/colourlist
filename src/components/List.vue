<template>
  <section>
    <ul>
      <li
        v-for='colour in colours'
        :style='{ backgroundColor: colour }'
        :key='colour'
      >{{colour}}</li>
    </ul>
  </section>
</template>

<script>
// Use this:
// http://jsfiddle.net/shanfan/ojgp5718/

export default {
  name: 'List',

  props: {
    colours: {
      type: Array,
      default: () => []
    }
  }
}
</script>
