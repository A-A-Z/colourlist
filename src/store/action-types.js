
// colour list actions
export const ON_NEW_COLOUR_INPUT = 'ON_NEW_COLOUR_INPUT'
export const ON_COLOUR_SUBMIT = 'ON_COLOUR_SUBMIT'

// cloud
export const CONNECT = 'CONNECT'
export const CREATE = 'CREATE'
export const LOAD = 'LOAD'
export const UPDATE = 'UPDATE'
export const GENERATE_PROJECT_ID = 'GENERATE_PROJECT_ID'

// project
export const ON_TITLE_CHANGE = 'ON_TITLE_CHANGE'
