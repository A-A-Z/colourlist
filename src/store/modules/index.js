export {default as colourList} from './colourList.js';
export {default as settings} from './settings.js';
export {default as cloud} from './cloud.js';
