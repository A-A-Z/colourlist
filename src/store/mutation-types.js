
// colour list
export const ADD_COLOUR_TO_LIST = 'ADD_COLOUR_TO_LIST'
export const NEW_COLOUR_CHANGE = 'NEW_COLOUR_CHANGE'
export const REMOVE_COLOUR_FROM_LIST = 'REMOVE_COLOUR_FROM_LIST'
export const SET_ACTIVE_COLOUR = 'SET_ACTIVE_COLOUR'
