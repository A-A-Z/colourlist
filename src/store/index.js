import Vue from 'vue'
import Vuex from 'vuex'
import { colourList, settings, cloud, project } from './modules'

Vue.use(Vuex)

export default new Vuex.Store({
  modules: {
    colourList,
    settings,
    cloud,
    project
  }
})
