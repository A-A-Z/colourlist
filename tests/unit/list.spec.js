import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import List from '@/components/List.vue'
import { colourList } from '@/store/modules'
import { SET_ACTIVE_COLOUR, SET_COLOURS } from '@/store/mutation-types.js'

Vue.use(Vuex)

describe('List.vue', () => {
  let wrapper
  let store

  beforeEach(() => {
    store = new Vuex.Store({ modules: { colourList } })

    wrapper = shallowMount(List, {
      store
    })
  })

  it('renders List section', () => {
    expect(wrapper.findAll('.o-colour-list')).toHaveLength(1)
  })

  it('renders List with zero colours', () => {
    expect(wrapper.findAll('.o-colour-list__item')).toHaveLength(0)
  })

  it('renders List with three colours', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ff0000', '#00ff00', '#0000ff'] })
    expect(wrapper.findAll('.o-colour-list__item')).toHaveLength(3)
  })

  it('renders colour label', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ff0000'] })
    expect(wrapper.find('.o-colour-list__label').text()).toBe('Red')
  })

  it('clicking on item makes colour active', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ffffff'] })
    wrapper
      .find('.o-colour-list__item')
      .trigger('click')

    expect(store.state.colourList.activeColour).toEqual('#ffffff')
  })

  it('hitting enter on item makes colour active', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ffffff'] })
    wrapper
      .find('.o-colour-list__item')
      .trigger('keyup.enter')

    expect(store.state.colourList.activeColour).toEqual('#ffffff')
  })

  it('hitting space on item makes colour active', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ffffff'] })
    wrapper
      .find('.o-colour-list__item')
      .trigger('keyup.space')

    expect(store.state.colourList.activeColour).toEqual('#ffffff')
  })

  it('having an active colour renders that colour as active', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ff0000', '#00ff00', '#0000ff'] })
    store.commit(`colourList/${SET_ACTIVE_COLOUR}`, '#00ff00')
    expect(wrapper.findAll('.o-colour-list__item--active')).toHaveLength(1)
  })

  // TODO Why no work!?
  // it('having an active colour shows the delete icon for that colour', () => {
  //   store.commit(`colourList/${SET_COLOURS}`, { list: ['#ff0000', '#00ff00', '#0000ff'] })
  //   store.commit(`colourList/${SET_ACTIVE_COLOUR}`, '#00ff00')
  //   expect(wrapper.findAll('.o-colour-list__delete')).toHaveLength(1)
  // })

  it('renders correctly', () => {
    store.commit(`colourList/${SET_COLOURS}`, { list: ['#ff0000', '#00ff00', '#0000ff'] })
    store.commit(`colourList/${SET_ACTIVE_COLOUR}`, '#00ff00')
    expect(wrapper.element).toMatchSnapshot()
  })

  //
  // it('clicking on item makes colour active', () => {
  //   wrapper.setProps({
  //     colours: [ '#ffffff' ],
  //     setActiveColour: jest.fn()
  //   })
  //   wrapper
  //     .find('.o-colour-list__item')
  //     .trigger('click')
  //
  //   expect(wrapper.vm.setActiveColour).toHaveBeenCalledWith('#ffffff')
  // })
  //

  //

  //
  // it('having an active colour shows the delete icon for that colour', () => {
  //   wrapper.setProps({
  //     colours: [
  //       '#111111',
  //       '#222222',
  //       '#333333'
  //     ],
  //     activeColour: '#222222'
  //   })
  //   expect(wrapper.findAll('.o-colour-list__item--active .o-colour-list__delete')).toHaveLength(1)
  // })
  //
  // it('clicking on delete icon calls deleteSavedColour function for that colour', () => {
  //   wrapper.setProps({
  //     colours: [
  //       '#111111',
  //       '#222222',
  //       '#333333'
  //     ],
  //     activeColour: '#222222',
  //     deleteSavedColour: jest.fn()
  //   })
  //   wrapper
  //     .find('.o-colour-list__delete')
  //     .trigger('click')
  //
  //   expect(wrapper.vm.deleteSavedColour).toHaveBeenCalledWith('#222222')
  // })
  //
  // it('computed coloursSorted sorts colours by hue', () => {
  //   wrapper.setProps({
  //     colours: [
  //       '#9999ff',
  //       '#ff0000',
  //       '#00ff00',
  //       '#5555ff',
  //       '#ff5555',
  //       '#0000ff',
  //       '#99ff99',
  //       '#ff9999',
  //       '#55ff55'
  //     ]
  //   })
  //   expect(wrapper.vm.coloursSorted).toEqual([
  //     '#ff9999',
  //     '#ff5555',
  //     '#ff0000',
  //     '#99ff99',
  //     '#55ff55',
  //     '#00ff00',
  //     '#9999ff',
  //     '#5555ff',
  //     '#0000ff'
  //   ])
  // })
})
