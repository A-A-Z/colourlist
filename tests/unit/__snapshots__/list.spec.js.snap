// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue renders correctly 1`] = `
<section
  class="t-main__section"
>
  <ul
    class="o-colour-list"
  >
    <li
      class="o-colour-list__item"
      role="button"
      style="background-color: rgb(255, 0, 0);"
      tabindex="0"
    >
      <!---->
       
      <div
        class="o-colour-list__label"
      >
        Red
      </div>
    </li>
    <li
      aria-pressed="true"
      class="o-colour-list__item o-colour-list__item--active"
      role="button"
      style="background-color: rgb(0, 255, 0);"
      tabindex="0"
    >
      <!---->
       
      <div
        class="o-colour-list__label"
      >
        Green
      </div>
    </li>
    <li
      class="o-colour-list__item"
      role="button"
      style="background-color: rgb(0, 0, 255);"
      tabindex="0"
    >
      <!---->
       
      <div
        class="o-colour-list__label"
      >
        Blue
      </div>
    </li>
  </ul>
</section>
`;
